srcts/src/components/toast.ts

```diff
@@ -0,0 +1,355 @@
+    let container = this.containers.get(position);
+
+    if (!container || !document.body.contains(container)) {
+      container = this._createToaster(position);
+      this.containers.set(position, container);
+    }
+
+    return container;
+  }
+
+  /**
+   * Creates a new toast container (toaster) DOM element for the specified
+   * position.
+   *
+   * @param position - The toast position to create a container for
+   * @returns A new DOM container element positioned and styled for toasts
+   * @private
+   */
+  private _createToaster(position: ToastPosition): HTMLElement {
+    const toaster = document.createElement("div");
+    toaster.className = "toast-container position-fixed p-1 p-md-2";
+    toaster.setAttribute("data-bslib-toast-container", position);
+    toaster.classList.add(...this._positionClasses(position));
+
+    document.body.appendChild(toaster);
+
+    return toaster;
+  }
+
+  /**
+   * Maps toast positions to their corresponding Bootstrap utility classes.
+   *
+   * @param position - The toast position
+   * @returns Array of CSS class names for positioning the container
+   * @private
+   */
+  private _positionClasses(position: ToastPosition): string[] {
+    const classMap: { [key in ToastPosition]: string[] } = {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "top-left": ["top-0", "start-0"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "top-center": ["top-0", "start-50", "translate-middle-x"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "top-right": ["top-0", "end-0"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "middle-left": ["top-50", "start-0", "translate-middle-y"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "middle-center": ["top-50", "start-50", "translate-middle"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "middle-right": ["top-50", "end-0", "translate-middle-y"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "bottom-left": ["bottom-0", "start-0"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "bottom-center": ["bottom-0", "start-50", "translate-middle-x"],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "bottom-right": ["bottom-0", "end-0"],
+    };
+
+    return classMap[position];
+  }
+}
+
+const toasterManager = new ToasterManager();
+
+/**
+ * Manages the lifecycle and state of an individual toast notification.
+ *
+ * Encapsulates all toast-specific behavior including progress bar animation,
+ * hover pause/resume functionality, and Bootstrap Toast integration.
+ */
+class BslibToastInstance {
+  private element: HTMLElement;
+  private bsToast: typeof bootstrapToast.prototype;
+  private progressBar: HTMLElement | null = null;
+  private startTime = 0;
+  private duration = 0;
```

Comment:
Part of me wants to call this something like `timeRemaining`?

Reply:
bca00b206907396af3a4bb0d4503125d5f8cce04